pkg/issuer/acme/dns/dns.go

```diff
@@ -36,7 +36,7 @@ type solver interface {
 // It is useful for mocking out a given provider since an alternate set of
 // constructors may be set.
 type dnsProviderConstructors struct {
-	cloudDNS   func(project string, serviceAccount []byte) (*clouddns.DNSProvider, error)
+	cloudDNS   func(project string, serviceAccountFile string, serviceAccount []byte) (*clouddns.DNSProvider, error)
```

Comment:
also pass through `ambient bool` here (same pattern as for route53), so we can check it's enabled